docs/unit_tests.rst

```diff
@@ -0,0 +1,126 @@
+============
+ Unit Tests
+============
+
+The Goals of Unit Testing
+=========================
```

Comment:
Can you match the headers to [the scheme used by the other rst files](https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#sections)?